packages/jaeger-ui-components/src/TracePageHeader/TracePageHeader.test.js

```diff
@@ -12,75 +12,120 @@
-    expect(wrapper.find({ traceName: getTraceName(trace.spans) })).toBeTruthy();
+    expect(screen.queryByRole('banner')).not.toBeInTheDocument();
+    expect(screen.queryByRole('heading', { traceName: getTraceName(trace.spans) })).not.toBeInTheDocument();
+    expect(screen.queryAllByRole('listitem')).toHaveLength(0);
+    expect(screen.queryByText(/Reset Selection/)).not.toBeInTheDocument();
   });
 
-  it('renders the header items', () => {
-    wrapper.find('.horizontal .item').forEach((item, i) => {
-      expect(item.contains(HEADER_ITEMS[i].title)).toBeTruthy();
-      expect(item.contains(HEADER_ITEMS[i].renderer(defaultProps.trace))).toBeTruthy();
-    });
+  it('should render the trace title', () => {
+    setup();
+
+    expect(screen.getByRole('heading', { traceName: getTraceName(trace.spans) })).toBeInTheDocument();
   });
 
-  it('renders start time in header with millisecond precision', () => {
-    const startTimeMs = wrapper.find(LabeledList).props().items[0].value.props.children[1].props.children;
-    expect(startTimeMs).toMatch(/:\d\d\.\d\d\d/g);
+  it('should render the header items', () => {
+    setup();
+
+    const headerItems = screen.queryAllByRole('listitem');
+
+    expect(headerItems).toHaveLength(5);
+    //                                                        Year-month-day hour-minute-second
+    expect(headerItems[0].textContent.match(/Trace Start:\d{4}-\d{2}-\d{2}\s\d{2}:\d{2}:\d{2}\.\d{3}/g)).toBeTruthy();
+    expect(headerItems[1].textContent.match(/Duration:[\d|\.][\.|\d|s][\.|\d|s]?[\d]?/)).toBeTruthy();
+    expect(headerItems[2].textContent.match(/Services:\d\d?/g)).toBeTruthy();
+    expect(headerItems[3].textContent.match(/Depth:\d\d?/)).toBeTruthy();
+    expect(headerItems[4].textContent.match(/Total Spans:\d\d?\d?\d?/)).toBeTruthy();
   });
 
-  it('renders a <SpanGraph>', () => {
-    expect(wrapper.find(SpanGraph).length).toBe(1);
+  it('should render a <SpanGraph>', () => {
+    setup();
+
+    expect(screen.getByText(/Reset Selection/)).toBeInTheDocument();
   });
 
   describe('observes the visibility toggles for various UX elements', () => {
     it('hides the minimap when hideMap === true', () => {
-      expect(wrapper.find(SpanGraph).length).toBe(1);
-      wrapper.setProps({ hideMap: true });
-      expect(wrapper.find(SpanGraph).length).toBe(0);
+      setup({ hideMap: true });
+
+      expect(screen.queryByText(/Reset Selection/)).not.toBeInTheDocument();
     });
 
     it('hides the summary when hideSummary === true', () => {
-      expect(wrapper.find(LabeledList).length).toBe(1);
-      wrapper.setProps({ hideSummary: true });
-      expect(wrapper.find(LabeledList).length).toBe(0);
+      const { rerender } = setup({ hideSummary: false });
+      expect(screen.queryAllByRole('listitem')).toHaveLength(5);
+
+      rerender(<TracePageHeader hideSummary={false} trace={null} />);
+      expect(screen.queryAllByRole('listitem')).toHaveLength(0);
```

Comment:
Becuase I was rerendering the component inline without the full props defined in setup the trace was defaulting to null, so it wasnt rendering anything, I left this in as an example.